lib/active_model/serializer/collection_serializer.rb

```diff
@@ -10,8 +10,14 @@ class CollectionSerializer
       def initialize(resources, options = {})
         @root = options[:root]
```

Comment:
I'm thinking this should really be `@root = collection_root(options[:root], options[:serializer])`

and something like (assuming root and json_key should behave the same... I don't remember)

```ruby
private

def collection_root(explicit_root, explicit_each_serializer)
  explicit_root || derived_root(explicit_each_serializer)
end

def derived_root(explicit_each_serializer)
 serializers.first.try!(:json_key).try!(:pluralize) ||
    (resources.empty && explicit_each_serializer.try!(:allocate).try!(:json_key).try!(:pluralize)) ||
    object.try!(:name).try!(:underscore).try!(:pluralize)
end
```

aside: all these `try` make my head hurt